### PR TITLE
Replace status column from placements index with provider column

### PR DESCRIPTION
**Pull request by dp-daly**
## Context

- Change third column from 'status' to 'provider' on the placements index of a given school making it clearer to users whether a provider is still to be assigned to the placement or not.
- This change will inform the design of any ‘manage placement’ functionality we may put in.

## Changes proposed in this pull request

- Remove the status column from the placements table on the placements index page
- Add the Provider column to the placements table on the placements index page
- The provider column should read “Not yet known” if no provider is present
- Update all “Not yet known” text to use the secondary colour for the placements table

## Guidance to review

- Review `app/views/placements/schools/placements/_table.html.erb`
- Review helper methods in TableHelper: `app/helpers/table_helper.rb`
- Run`bundle exec rspec spec/system/placements/support/schools/placements/support_user_views_placements_spec.rb`  
- Run `bundle exec rspec spec/system/placements/schools/placements/view_placements_list_spec.rb`

## Link to Trello card

https://trello.com/c/aC2EAnrS/661-add-provider-column-to-school-placements-index

## Screenshots

Screenshot 2024-08-13 at 14 28 06

### CODE REVIEW

File: app/assets/stylesheets/components/_typography.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.secondary-text {
+    color: $govuk-secondary-text-colour;
+}
```
Comment (Kizr) [inline app/assets/stylesheets/components/_typography.scss:3]: Please can we add a new line to this file?

File: app/helpers/table_helper.rb [hunk 2]
```diff
@@ -4,4 +4,12 @@ def app_table(*args, **kwargs, &block)
...
     end
   end
+
+  def provider_colour(placement)
+    placement.provider_name == I18n.t("placements.schools.placements.not_yet_known") ? "secondary-text" : ""
+  end
+
+  def mentor_colour(placement)
+    placement.mentor_names == I18n.t("placements.schools.placements.not_yet_known") ? "secondary-text" : ""
+  end
```
Comment (Kizr) [inline app/helpers/table_helper.rb:14]: Could we combine these into a single method? Could the method name be more descriptive if we do this?

**Review by Kizr — DISMISSED** [review]

File: app/helpers/table_helper.rb [hunk 3]
```diff
@@ -4,4 +4,8 @@ def app_table(*args, **kwargs, &block)
       govuk_table(*args, **kwargs, &block)
     end
   end
+
+  def make_not_yet_known_secondary_colour(placement)
```
Comment (Kizr) [inline app/helpers/table_helper.rb:8]: Nice work, this title is descriptive! A possible change to reduce the cognitive load would be `not_yet_known_colour_class`, we don't want to couple the name to the specific method logic too tightly!

File: app/helpers/table_helper.rb [hunk 4]
```diff
@@ -4,4 +4,8 @@ def app_table(*args, **kwargs, &block)
...
+    placement.provider_name == I18n.t("placements.schools.placements.not_yet_known") || placement.mentor_names == I18n.t("placements.schools.placements.not_yet_known") ? "secondary-text" : ""
+  end
```
Comment (Kizr) [inline app/helpers/table_helper.rb:10]: We may be able to refactor this further by changing the argument slightely, how do you feel about something like this?

```ruby
def table_text_class(data)
  data == "Not yet known" ? "secondary-text" : ""
end

table_data_class(placement.provider_name)
table_data_class(placement.mentor_names)
```

**Review by JamieCleare2525 — APPROVED** [review]
LGTM!

**Pull request merged**